File: Stock Span/stock_span.py
def stock_span(prices):
    spans = []
    prevs = []
    for price in prices:
        days = 1
        while prevs and prevs[-1][0] <= price:
            days += prevs.pop()[1]
        prevs.append((price, days))
        spans.append(days)
    return spans

a = [100,80,60,70, 60,75,85]
for x in stock_span(a):
    print(x, end=" ")

File: Stock Span/StockSpan.kt
fun stockSpan(prices: List<Int>): MutableList<Int> {
    val spans = mutableListOf<Int>()
    val prevs = mutableListOf<Pair<Int, Int>>()
    for (price in prices) {
        var days = 1
        while (!prevs.isEmpty() && (prevs.last().first <= price)) {
            days += prevs.removeLast().second
        }
        prevs.add(Pair(price, days))
        spans.add(days)
    }
    return spans
}

fun main() {
    val a = listOf(100,80,60,70, 60,75,85)
    for (x in stockSpan(a)) {
        print("$x ")
    }
}
